psi4/driver/qcdb/libmintsmolecule.py

```diff
@@ -1821,44 +1869,55 @@ def inertia_tensor(self):
+
+        Notes
+        -----
+        This used to return a list with inf values as None.
+
+        """
+        import numpy as np
 
         evals, evecs = diagonalize3x3symmat(self.inertia_tensor())
         evals = sorted(evals)
-        isLinear = True if evals[0] < ZERO else False
-        isOne = True if evals[1] < ZERO else False
-        isAtom = True if evals[2] < ZERO else False
+        evals = np.asarray(evals)
 
         im_amuA = psi_bohr2angstroms * psi_bohr2angstroms
-        im_ghz = psi_h * psi_na * 1E14 / (8.0 * math.pi * math.pi * psi_bohr2angstroms * psi_bohr2angstroms)
-        im_mhz = im_ghz * 1000
-        im_cm = im_ghz * 1E7 / psi_c
-
+        im_ghz = psi_h * psi_na * 1e14 / (8 * math.pi * math.pi * psi_bohr2angstroms * psi_bohr2angstroms)
+        im_mhz = im_ghz * 1000.
+        im_cm = im_ghz * 1.e7 / psi_c
+
+        rc_moi = {}
+        rc_moi['u a0^2'] = evals
+        rc_moi['u A^2'] = evals * im_amuA
+        with np.errstate(divide='ignore'):
+            rc_moi['GHz'] = im_ghz / evals
+            rc_moi['MHz'] = im_mhz / evals
+            rc_moi['cm^-1'] = im_cm / evals
+
+        fmt = """  {:12}    {a:3} {:16.8f}    {b:3} {:16.8f}    {c:3} {:16.8f}\n"""
         text = "        Moments of Inertia and Rotational Constants\n\n"
-        text += '  %-12s    %3s %16.8f    %3s %16.8f    %3s %16.8f\n' % \
-            ('[amu B^2]', 'I_A', evals[0], 'I_B', evals[1], 'I_C', evals[2])
-        text += '  %-12s    %3s %16.8f    %3s %16.8f    %3s %16.8f\n' % \
-            ('[amu A^2]', 'I_A', evals[0] * im_amuA, 'I_B', evals[1] * im_amuA, 'I_C', evals[2] * im_amuA)
-        text += '  %-12s    %3s %16s    %3s %16s    %3s %16s\n' % ('[GHz]', \
-            'A', '%16.8f' % (im_ghz / evals[0]) if not isLinear else '*****', \
-            'B', '%16.8f' % (im_ghz / evals[1]) if not isOne else '*****', \
-            'C', '%16.8f' % (im_ghz / evals[2]) if not isAtom else '*****')
-        text += '  %-12s    %3s %16s    %3s %16s    %3s %16s\n' % ('[MHz]', \
-            'A', '%16.8f' % (im_mhz / evals[0]) if not isLinear else '*****', \
-            'B', '%16.8f' % (im_mhz / evals[1]) if not isOne else '*****', \
-            'C', '%16.8f' % (im_mhz / evals[2]) if not isAtom else '*****')
-        text += '  %-12s    %3s %16s    %3s %16s    %3s %16s\n' % ('[cm^-1]', \
-            'A', '%16.8f' % (im_cm / evals[0]) if not isLinear else '*****', \
-            'B', '%16.8f' % (im_cm / evals[1]) if not isOne else '*****', \
-            'C', '%16.8f' % (im_cm / evals[2]) if not isAtom else '*****')
+        text += fmt.format('[u a0^2]', a='I_A', b='I_B', c='I_C', *rc_moi['u a0^2'])
+        text += fmt.format('[u A^2]', a='I_A', b='I_B', c='I_C', *rc_moi['u A^2'])
+        text += fmt.format('[GHz]', a='A', b='B', c='C', *rc_moi['GHz'])
+        text += fmt.format('[MHz]', a='A', b='B', c='C', *rc_moi['MHz'])
+        text += fmt.format('[cm^-1]', a='A', b='B', c='C', *rc_moi['cm^-1'])
         print(text)
```

Comment:
Is this missing a redirect to the outfile, or deliberately going to stdout?

Reply:
Pretty much the whole of qcdb outside those pieces that would clutter up a psi4 calc has a serious outfile/stdout problem for which I don't have a clean solution. Yes, it's just heading to stdout.

Reply:
We can always overwrite stdout to psi4 output (I think folks would be unhappy). Or provide a output handle to the functions.

Reply:
I'd really like a `logging` type approach where an output resource could be global. But an output handle is probably more realistic. I've got a lot of `goodie, text = function_of_worth()` going now, which complicates function returns. I may be searching the wrong things, but I haven't seen a good sol'n out there for py modules.